plugins/scaffolder/src/components/MultistepJsonForm/MultistepJsonForm.tsx

```diff
@@ -60,71 +64,37 @@ type Props = {
   fields?: FormProps<any>['fields'];
   finishButtonLabel?: string;
   layouts: LayoutOptions[];
+  ReviewStepComponent?: ComponentType<ReviewStepProps>;
 };
 
-export function getUiSchemasFromSteps(steps: Step[]): UiSchema[] {
-  const uiSchemas: Array<UiSchema> = [];
-  steps.forEach(step => {
-    const schemaProps = step.schema.properties as JsonObject;
-    for (const key in schemaProps) {
-      if (schemaProps.hasOwnProperty(key)) {
-        const uiSchema = schemaProps[key] as UiSchema;
-        uiSchema.name = key;
-        uiSchemas.push(uiSchema);
-      }
-    }
-  });
-  return uiSchemas;
-}
-
-export function getReviewData(formData: Record<string, any>, steps: Step[]) {
-  const uiSchemas = getUiSchemasFromSteps(steps);
-  const reviewData: Record<string, any> = {};
-  for (const key in formData) {
-    if (formData.hasOwnProperty(key)) {
-      const uiSchema = uiSchemas.find(us => us.name === key);
-
-      if (!uiSchema) {
-        reviewData[key] = formData[key];
-        continue;
-      }
-
-      if (uiSchema['ui:widget'] === 'password') {
-        reviewData[key] = '******';
-        continue;
-      }
-
-      if (!uiSchema['ui:backstage'] || !uiSchema['ui:backstage'].review) {
-        reviewData[key] = formData[key];
-        continue;
-      }
-
-      const review = uiSchema['ui:backstage'].review as JsonObject;
-      if (review.mask) {
-        reviewData[key] = review.mask;
-        continue;
-      }
-
-      if (!review.show) {
-        continue;
-      }
-      reviewData[key] = formData[key];
-    }
-  }
-
-  return reviewData;
+export function getSchemasFromSteps(steps: Step[]): {
+  uiSchema: UiSchema;
+  mergedSchema: JsonObject;
+  schema: JsonObject;
+}[] {
+  return steps.map(({ schema }) => ({
+    mergedSchema: schema,
+    ...extractSchemaFromStep(schema),
+  })) as unknown as {
```

Comment:
Just one last thing here, what the reason for this `unknown` cast here?

Reply:
Yeah, generics are messed up, in one place it's any and other is more fine grained defined.

Could be a good input to create a separate story to investigate what where went wrong :) 

<img width="1161" alt="Screenshot 2022-11-17 at 16 39 36" src="https://user-images.githubusercontent.com/2265094/202492279-d48c71e3-0311-4150-93f4-9889a6a8da68.png">

Reply:
🙄